lightning-invoice/src/utils.rs

```diff
@@ -21,6 +21,98 @@ use core::convert::TryInto;
 use core::ops::Deref;
 use core::time::Duration;
 
+#[cfg(feature = "std")]
+/// Utility to create an invoice that can be paid to one of multiple nodes, or a "phantom invoice."
+/// This works because the invoice officially pays to a fake node (the "phantom"). Useful for
+/// load-balancing payments between multiple nodes.
```

Comment:
I'm not sure if you have strong load-balancing guarantees offered to the receiver. A phantom invoice receiver is always free to select any node from the set of participating nodes based on which one is the shortest/cheaper to reach. However, the shortest to reach from the sender maybe the node with the most depleted/congestioned channels from the receiver viewpoint.

So I see more phantom invoices as a payment routing cost optimization, if the receiver has deployed node equidistantly in the routing graph ?

Reply:
I mean its still load-balancing, just not *guaranteed* load balancing. No different from people who load-balance with DNS RR and other schemes, it'll balance somewhat, but nothing is guaranteed. We still call that load-balancing.